### PR TITLE
PEAR-1822: Files table in case summary

### PR DESCRIPTION
**Pull request by paribartandhakal**
## Checklist
- [ ] Added proper unit tests
- [ ] Left proper TODO messages for any remaining tasks
- [ ] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

## Screenshots/Screen Recordings (if Appropriate)
Screenshot 2024-03-05 at 3 55 32 PM

Screenshot 2024-03-06 at 10 18 17 AM

### CODE REVIEW

File: packages/portal-proto/src/features/cDave/CDaveCard/QQPlot.tsx [hunk 1]
```diff
@@ -175,7 +175,7 @@ const QQPlot: React.FC<QQPlotProps> = ({
         },
       ],
     }),
-    [chartData, color, lineData, label],
+    [chartData, color, lineData, label, width],
```
Comment (amlehman) [inline packages/portal-proto/src/features/cDave/CDaveCard/QQPlot.tsx:178]: Thanks for fixing this 🙏

File: packages/portal-proto/src/features/cases/FilesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+import { mapGdcFileToCartFile } from "../files/utils";
+import download from "@/utils/download";
+import { downloadTSV } from "@/components/Table/utils";
+import { convertDateToString } from "@/utils/date";
+
+interface FilesTableProps {
+  caseId: string;
+}
+
+type CaseFilesTableDataType = {
```
Comment (amlehman) [inline packages/portal-proto/src/features/cases/FilesTable.tsx]: Could you define this as a pick type of GdcFile like this? https://github.com/NCI-GDC/gdc-frontend-framework/blob/ca5032df7da7d503631d1ee7b60c52175fef5bd2/packages/core/src/features/cart/updateCartSlice.ts#L6

**Review by amlehman — APPROVED** [review]

**Comment by wteouchicago** [comment]
Small issues:
1. tooltip/text displayed when the search field is active isn't displayed when the case summary page is loaded in a modal
2. TSV name should be files-table.YYYY-MM-DD.tsv (I don't love it either, but it's to be consistent with other files table)

nice work! feel free to merge when the changes are made and it passes code review

**Review by craigrbarnes — CHANGES_REQUESTED** [review]
Looks fine to me. Please resolve @wteouchicago comment

File: packages/portal-proto/src/features/cases/FilesTable.tsx (hunk 2 above)
Comment (paribartandhakal) [reply packages/portal-proto/src/features/cases/FilesTable.tsx]: This is changed.

**Comment by paribartandhakal** [comment]
> Looks fine to me. Please resolve @wteouchicago comment

@craigrbarnes Wendy's comments have been addressed. I have updated the screenshots too.

**Comment by wteouchicago** [comment]
Issues 1-2 look fixed.

Issues:
3. In the Customize Columns feature, the "File UUID" column is named "File ID" instead of "File UUID". Sorry if I missed this earlier.
4. The File UUID column should not be displayed by default. (Did I miss all these earlier or did the latest commit change these?)

**Comment by paribartandhakal** [comment]
> Issues 1-2 look fixed.
> 
> Issues: 3. In the Customize Columns feature, the "File UUID" column is named "File ID" instead of "File UUID". Sorry if I missed this earlier. 4. The File UUID column should not be displayed by default. (Did I miss all these earlier or did the latest commit change these?)

You never miss things 😄 
It was because of my changes and I fixed it now.

**Comment by wteouchicago** [comment]
looks good - all fixed. 👍

**Review by craigrbarnes — APPROVED** [review]

**Pull request merged**